action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -102,10 +103,18 @@ func AsyncContractTrieOption() StateDBAdapterOption {
 	}
 }
 
-// UsePendingNonceOption set usePendingNonce as true
-func UsePendingNonceOption() StateDBAdapterOption {
+// UseConfirmedNonceOption set usePendingNonce as true
+func UseConfirmedNonceOption() StateDBAdapterOption {
 	return func(adapter *StateDBAdapter) error {
-		adapter.usePendingNonce = true
+		adapter.useConfirmedNonce = true
+		return nil
+	}
+}
+
+// LegacyNonceAccountOption set legacyNonceAccount as true
+func LegacyNonceAccountOption() StateDBAdapterOption {
+	return func(adapter *StateDBAdapter) error {
+		adapter.legacyNonceAccount = true
```

Comment:
Are `legacyNonceAccount` and `useConfirmedNonce` independent?

Reply:
we expect: if `legacyNonceAccount == false`, then `useConfirmedNonce == false`.